### PR TITLE
Use `slices.Compact` instead of own logic

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
The `slices` package can be used to remove duplicates.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/validate/image.go [hunk 1]
```diff
@@ -221,7 +220,7 @@ func pullImageList(c internalapi.ImageManagerService, imageList []string, podCon
 	for _, imageName := range imageList {
 		ids = append(ids, framework.PullPublicImage(c, imageName, podConfig))
 	}
-	return ids
+	return slices.Compact(ids)
```
Comment (SergeyKanzhelev) [inline pkg/validate/image.go:223]: should we sort before compact? It looks like docs are saying it is deduping consequetive values, not all unique values

Comment (saschagrunert) [reply pkg/validate/image.go:224]: That's tricky, the tests expect the ID's in the same order as the images, but I guess we can sort the image list and dedup the IDs then.

Comment (saschagrunert) [reply pkg/validate/image.go]: Yeah, that depends on the test. I'll fix that up.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/appeove

**Pull request merged**